src/test/versionbits_tests.cpp

```diff
@@ -226,6 +267,13 @@ BOOST_AUTO_TEST_CASE(versionbits_test)
         // Make sure that no deployment tries to set an invalid bit.
         BOOST_CHECK_EQUAL(bitmask & ~(uint32_t)VERSIONBITS_TOP_MASK, bitmask);
 
+        // Check min_activation_height is on a retarget boundary
+        BOOST_CHECK_EQUAL(mainnetParams.vDeployments[i].min_activation_height % mainnetParams.nMinerConfirmationWindow, 0);
```

Comment:
This can be addressed in a followup:
```bash
warning: comparison of integers of different signs: 'const unsigned int' and 'const int' [-Wsign-compare]
...
test/versionbits_tests.cpp:271:9: note: in instantiation of function template specialization 'boost::test_tools::tt_detail::check_frwd<boost::test_tools::tt_detail::equal_impl_frwd, unsigned int, int>' requested here
        BOOST_CHECK_EQUAL(mainnetParams.vDeployments[i].min_activation_height % mainnetParams.nMinerConfirmationWindow, 0);
```

Reply:
I can't reproduce (#19123), but the warning seems correct:

```cpp
#include <iostream>
#include <type_traits>
#include <cstdint>
 
int main()
{
  std::cout << std::boolalpha;

  std::cout << std::is_same_v<int     , decltype(int{1} % unsigned{2})> << '\n'; // false
  std::cout << std::is_same_v<unsigned, decltype(int{1} % unsigned{2})> << '\n'; // true
}
```

Reply:
Fixed in #21691